### PR TITLE
DOCSP-6633: Kafka connector guide - migration

### PR DESCRIPTION
**Pull request by ccho-mongodb**
JIRA: https://jira.mongodb.org/browse/DOCSP-6633

Stage: https://docs-mongodborg-staging.corp.mongodb.com/ecosystem/docsworker/DOCSP-6633-kafka-migration/connectors/kafka-connect-migration.html

### CODE REVIEW

File: source/connectors/kafka-connect-migration.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+This section lists the changes needed to migrate from the deprecated,
+community-developed version of the Kafka Connector (hosted in the
+`hpgrahsl/kafka-connect-mongodb GitHub repository
+<https://github.com/hpgrahsl/kafka-connect-mongodb>`_) to the `official
+connector <https://github.com/mongodb/mongo-kafka>`_.
+
+Once your project build file is updated to reference the official Kafka
+Connect library, use the following information to migrate your
+project.
```
Comment (nlarew) [inline source/connectors/kafka-connect-migration.txt]: This seems like an important step. How can I find info on what to change in my build file?

File: source/connectors/kafka-connect-migration.txt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+Package and Class Name Changes
+------------------------------
```
Comment (nlarew) [inline source/connectors/kafka-connect-migration.txt]: This feels like a procedural step: **Update Connector Package and Class Names**

File: source/connectors/kafka-connect-migration.txt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  ``at.grahsl.kafka.connect.mongodb`` to ``com.mongodb.kafka.connect``
+
+- The ``MongoDbSinkConnector`` class has been renamed ``MongoSinkConnector``
+
+- The ``MongoDbSinkConnectorConfig`` class has been split into
+  ``sink.MongoSinkConfig`` and ``sink.MongoSinkTopicConfig`` classes. Refer to
+  the source for a complete list of properties in `MongoSinkConfig
+  <https://github.com/mongodb/mongo-kafka/blob/master/src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java>`_
+  and `MongoSinkTopicConfig
+  <https://github.com/mongodb/mongo-kafka/blob/master/src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java>`_.
```
Comment (nlarew) [inline source/connectors/kafka-connect-migration.txt]: I wonder if this section would be clearer as a table with **New** and **Old** columns or somesuch

File: source/connectors/kafka-connect-migration.txt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Configuration Changes
+---------------------
```
Comment (nlarew) [inline source/connectors/kafka-connect-migration.txt]: This feels like a procedural step: **Update Your Kafka Configuration**

Comment (nlarew) [reply source/connectors/kafka-connect-migration.txt]: It's not super clear what config these changes happen to. Is it a MongoDB config file or something in Kafka?

File: source/connectors/kafka-connect-migration.txt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  <https://github.com/mongodb/mongo-kafka/blob/master/docs/sink.md#topic-specific-configuration-settings>`_
+  for available per-topic and override options.
+
+- The ``document.id.strategies`` configuration setting key has been removed.
+  Instead, to include a strategy, implement the `IdStrategy
+  <https://github.com/mongodb/mongo-kafka/blob/master/src/main/java/com/mongodb/kafka/connect/sink/processor/id/strategy/IdStrategy.java>`_
+  interface.
+
+PostProcessor Changes
+---------------------
```
Comment (nlarew) [inline source/connectors/kafka-connect-migration.txt]: This feels like a procedural step: **Update PostProcessor Classes**

File: source/connectors/kafka-connect-migration.txt (hunk 1 above)
Comment (ccho-mongodb) [reply source/connectors/kafka-connect-migration.txt]: Would it be sufficient to reword this to say:

"Update any project dependencies that reference hpgrahsl/kafka-connect-mongodb to point to mongodb/mongo-kafka instead"
since there are several different tools the developer could be using such as Gradle, Maven, Ant, or other?

File: source/connectors/kafka-connect-migration.txt (hunk 5 above)
Comment (ccho-mongodb) [reply source/connectors/kafka-connect-migration.txt]: 👍

File: source/connectors/kafka-connect-migration.txt (hunk 2 above)
Comment (ccho-mongodb) [reply source/connectors/kafka-connect-migration.txt]: Thanks, will update.

File: source/connectors/kafka-connect-migration.txt (hunk 3 above)
Comment (ccho-mongodb) [reply source/connectors/kafka-connect-migration.txt]: I thought about that, but I'm not sure how I would represent items such as "remove the `mongodb.` prefix from all configurations" in a table like that other than attempting to identify every possible configuration that starts with `mongodb.` and creating a row for it.

**Comment by ccho-mongodb** [comment]
After reading more of the documentation, I think I mis-identified some of the settings and will fix them in the upcoming commit.

File: source/connectors/kafka-connect-migration.txt (hunk 4 above)
Comment (ccho-mongodb) [reply source/connectors/kafka-connect-migration.txt]: Appears to be both source and sink connector configuration files.
I think the scope of this task changed from reformatting to rewriting the guides late last week. I'll probably need a resource to clarify a number of questions I have due to the current state of the documentation.

File: source/connectors/kafka-connect-migration.txt (hunk 3 above)
Comment (nlarew) [reply source/connectors/kafka-connect-migration.txt]: True that would be a little tricky. I primarily meant this section though (Package and Class Name Changes), not the Configuration Changes section.
